status/condition_set.go

```diff
@@ -176,52 +177,47 @@ func (r ConditionSet) SetUnknown(conditionType string) {
 
 // recomputeRootCondition marks the root condition to true if all other dependents are also true.
 func (r ConditionSet) recomputeRootCondition(conditionType string) {
-	if condition := r.findUnhappyDependent(); condition != nil {
-		// Propagate unroot dependent to root condition.
+	if conditions := r.findUnhappyDependents(); len(conditions) > 0 {
+		// Propagate dependents to root condition.
 		r.Set(Condition{
-			Type:    r.root,
-			Status:  condition.Status,
-			Reason:  condition.Reason,
-			Message: condition.Message,
+			Type: r.root,
+			// Set false if any are false
+			Status: lo.Ternary(
+				lo.ContainsBy(conditions, func(condition Condition) bool { return condition.IsFalse() }),
+				metav1.ConditionFalse,
+				metav1.ConditionUnknown,
+			),
+			Reason: "UnhealthyDependent",
+			Message: strings.Join(lo.Map(conditions, func(condition Condition, _ int) string {
+				return fmt.Sprintf("%s=%s:%s", condition.Type, condition.Status, condition.Reason)
+			}), ","),
 		})
 	} else if conditionType != r.root {
 		// Set the root condition to true.
 		r.SetTrue(r.root)
 	}
 }
 
-func (c ConditionSet) findUnhappyDependent() *Condition {
+func (c ConditionSet) findUnhappyDependents() []Condition {
 	// This only works if there are dependents.
 	if len(c.dependents) == 0 {
 		return nil
 	}
 
 	// Get dependent conditions
-	conditions := lo.Filter(c.object.GetConditions(), func(condition Condition, _ int) bool {
+	conditions := c.object.GetConditions()
+	conditions = lo.Filter(conditions, func(condition Condition, _ int) bool {
```

Comment:
consider inlining
```suggestion
	conditions = lo.Filter(c.object.GetConditions(), func(condition Condition, _ int) bool {
```

Reply:
I actually did this intentionally as a data pipeline. Each command can just use `=` rather than the first being `:=`. Just an ole style thing I've come accustomed to